vector-im/element-android
# [Location sharing] Show own location in map views

## Type of change


## Content


Adding locate button in the expanded map views when looking at static or live location sharing.

## Motivation and context


Closes #8110 

## Screenshots / GIFs





Static location share
<img src="https://user-images.githubusercontent.com/46314705/219707218-c9d8016c-13b4-4b70-bc28-d08912ade1c0.png" width=300 />

Live location share
<img src="https://user-images.githubusercontent.com/46314705/219708501-756979bb-a59d-4ba1-8873-27c753c8ba55.png" width=300 />

## Tests



- Open a static location share message
- Check a locate button is displayed on the map
- Press the locate button
- Check the map is centered on the current user location
- Open a running live location share message
- Check a locate button is displayed on the map (only when current user is not live sharing their location)
- Press the locate button
- Check the map is centered on the current user location

## Tested devices

- OS version(s): Android 11

## Checklist
